### DISCUSSION TITLE
Ionic/Capacitor Rounds HTML Canvas coordinates in XCode build, not in dev server

### DISCUSSION THREAD

**Discussion by andrew-lyons** [ionic-team/capacitor#4991] [Q&A]
I am developing a capacitor plugin that requires floating point numbers from the client. I am using an HTML canvas and listening to the touch events associated.

When using ionic serve for the dev environment, my touchevent data looks like this:

```
12:38:06.813 Canvas.tsx:43 screenX :  762.3359375
12:38:06.813 Canvas.tsx:43 screenY :  756.05078125
12:38:06.813 Canvas.tsx:43 clientX :  201.3359375
12:38:06.813 Canvas.tsx:43 clientY :  585.05078125
12:38:06.813 Canvas.tsx:43 pageX :  201.3359375
12:38:06.813 Canvas.tsx:43 pageY :  585.05078125
12:38:06.814 Canvas.tsx:43 radiusX :  11.5
12:38:06.814 Canvas.tsx:43 radiusY :  11.5
12:38:06.814 Canvas.tsx:43 rotationAngle :  0
12:38:06.814 Canvas.tsx:43 force :  1
```

but in my XCode build, I log this:

```
⚡️  [log] - identifier :  1269311752
⚡️  [log] - clientX :  265
⚡️  [log] - clientY :  331
⚡️  [log] - pageX :  265
⚡️  [log] - pageY :  331
⚡️  [log] - screenX :  265
⚡️  [log] - screenY :  331
⚡️  [log] - radiusX :  21.15234375
⚡️  [log] - radiusY :  21.15234375
⚡️  [log] - rotationAngle :  0
⚡️  [log] - force :  0
⚡️  [log] - touchType :  direct
⚡️  [log] - altitudeAngle :  0
⚡️  [log] - azimuthAngle :  0
```
Notice the XCode results are integers, not floating point or doubles. I have no idea why this would be different between the dev environment and production build. Anyone have insights?

**Comment by jcesarmobile** [comment]
When you use ionic serve you mean you are running on desktop or you load the site on the phone?

I've added a touch listener to the body and on the iPhone I always get integers, doesn't matter if it's a website or a Capacitor app, while on desktop browsers and Android I get floating points.
